trestle/core/control_interface.py

```diff
@@ -646,15 +646,22 @@ def _param_label_choices_as_str(param: common.Parameter, verbose=False, brackets
 
     @staticmethod
     def _param_values_assignment_str(param: common.Parameter) -> str:
-        """Convert param label or choices to string, using choices if present."""
-        choices = ControlInterface._param_selection_as_str(param, True, False)
-        if choices:
-            text = f'[{choices}]'
-        elif param.label:
-            text = f'[Assignment: {param.label}]'
-        else:
-            text = param.id
-        return text
+        """Convert param values, label or choices to string."""
+        text = ''
+        # use values if present
+        param_str = ControlInterface._param_values_as_str(param, False)
+        if param_str:
+            text = f'{param_str}'
+        # otherwise use param selection if present
+        if not text:
+            param_str = ControlInterface._param_selection_as_str(param, True, False)
+            if param_str:
+                text = param_str
+        # use label and param_id as fallbacks
+        if not text:
+            text = param.label if param.label else param.id
+            text = f'Assignment: {text}'
+        return f'[{text}]'
```

Comment:
Text variable not needed?
```
param_str = ControlInterface._param_values_as_str(param, False)
if param_str:
    param_str = f'{param_str}'
# otherwise use param selection if present
else:
        param_str = ControlInterface._param_selection_as_str(param, True, False)
       # use label and param_id as fallbacks
      if not param_str:
          param_str = param.label if param.label else param.id
          param_str = f'Assignment: {text}'
return f'[{param_str}]'
         ```

Reply:
The function needed to be rewritten for the recent changes and incorporates your requested change.